fe/fe-core/src/main/java/com/starrocks/server/WarehouseManager.java

```diff
@@ -0,0 +1,212 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.server;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.gson.annotations.SerializedName;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.common.DdlException;
+import com.starrocks.common.io.Text;
+import com.starrocks.common.io.Writable;
+import com.starrocks.common.proc.BaseProcResult;
+import com.starrocks.common.proc.DbsProcDir;
+import com.starrocks.common.proc.ExternalDbsProcDir;
+import com.starrocks.common.proc.ProcDirInterface;
+import com.starrocks.common.proc.ProcNodeInterface;
+import com.starrocks.common.proc.ProcResult;
+import com.starrocks.persist.gson.GsonUtils;
+import com.starrocks.sql.ast.CreateWarehouseStmt;
+import com.starrocks.warehouse.Warehouse;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.DataInput;
+import java.io.DataInputStream;
+import java.io.DataOutput;
+import java.io.DataOutputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public class WarehouseManager implements Writable {
+    private static final Logger LOG = LogManager.getLogger(WarehouseManager.class);
+
+    @SerializedName(value = "idToWh")
+    private final ConcurrentHashMap<Long, Warehouse> idToWh = new ConcurrentHashMap<>();
```

Comment:
if we use rw lock, we don't need to use ConcurrentHashMap, normal hashMap is enough